application/shared-kernel/ApiCore/Endpoints/TrackEndpoints.cs

```diff
@@ -0,0 +1,195 @@
+
+                    telemetryClient.TrackMetric(telemetry);
+                    break;
+                }
+                case "RemoteDependencyData":
+                    // Ignore remote dependency data
+                    break;
+                default:
+                    logger.LogWarning($"Unsupported telemetry type: {trackRequestDto.Data.BaseType}");
+                    break;
+            }
+        }
+
+        return new TrackResponseSuccessDto(true, "Telemetry sent.");
+    }
+
+    private static IEnumerable<ExceptionDetailsInfo> GetExceptionDetailsInfos(TrackRequestDto trackRequestDto)
+    {
+        var exceptionDetailsInfos = trackRequestDto.Data.BaseData.Exceptions
+            .Select(exception => new ExceptionDetailsInfo(
+                0,
+                0,
+                exception.TypeName,
+                exception.Message,
+                exception.HasFullStack,
+                exception.Stack,
+                exception.ParsedStack.Select(parsedStack => new StackFrame(
+                    parsedStack.Assembly,
+                    parsedStack.FileName,
+                    parsedStack.Level,
+                    parsedStack.Line,
+                    parsedStack.Method
+                ))
+            ));
+        return exceptionDetailsInfos;
+    }
+
+    private static void CopyDictionary<TValue>(IDictionary<string, TValue>? source, IDictionary<string, TValue> target)
+    {
+        if (source == null) return;
+
+        foreach (var pair in source)
+        {
+            if (string.IsNullOrEmpty(pair.Key))
+            {
+                continue;
+            }
+
+            if (!target.ContainsKey(pair.Key))
+            {
+                target[pair.Key] = pair.Value;
+            }
+        }
+    }
+}
+
+internal abstract class TrackEndpointsLoggerCategory;
+
+[UsedImplicitly]
+public record TrackResponseSuccessDto(bool Success, string Message);
+
+[UsedImplicitly]
+public record TrackRequestDto(
+    string Time,
+    // ReSharper disable once InconsistentNaming
+    string IKey,
+    string Name,
+    Dictionary<string, string> Tags,
+    TrackRequestDataDto Data
+);
+
+[UsedImplicitly]
+public record TrackRequestDataDto(string BaseType, TrackRequestBaseDataDto BaseData);
+
+[UsedImplicitly]
+public record TrackRequestBaseDataDto(
+    string Name,
+    string Url,
+    string Duration,
+    Dictionary<string, string> Properties,
+    Dictionary<string, double> Measurements,
+    List<TrackRequestMetricsDto> Metrics,
+    List<TrackRequestExceptionDto> Exceptions,
+    SeverityLevel SeverityLevel,
+    string Id
+);
+
+[UsedImplicitly]
+public record TrackRequestExceptionDto(
+    string TypeName,
+    string Message,
+    bool HasFullStack,
+    string Stack,
+    List<TrackRequestParsedStackDto> ParsedStack
+);
+
+[UsedImplicitly]
+public record TrackRequestParsedStackDto(
+    string Assembly,
+    string FileName,
+    string Method,
+    int Line,
+    int Level
+);
```

Comment:
This can be one line:

```suggestion
[UsedImplicitly]
public record TrackRequestParsedStackDto(string Assembly, string FileName, string Method, int Line, int Level);
```